src/network/_Private/throw_socket_error.php

```diff
@@ -13,15 +13,12 @@
 use namespace HH\Lib\Experimental\{IO, Network, OS};
 use namespace HH\Lib\Str;
 
-function throw_socket_error(string $_operation, int $errno): noreturn {
-  invariant($errno !== 0, "%s should not be called on success", __FUNCTION__);
-  if ($errno < 0) {
-    throw new Network\HostResolutionException((-($errno + 10000)) as OS\HErrno);
-  }
-  switch ($errno as OS\Errno) {
-    case OS\Errno::EADDRNOTAVAIL:
-      throw new Network\AddressNotAvailableException();
-    default:
-      throw new IO\_Private\UnhandledOSErrnoException($errno as OS\Errno);
+function throw_socket_error(int $php_socket_error, string $message): noreturn {
+  invariant($php_socket_error !== 0, "%s should not be called on success", __FUNCTION__);
+  if ($php_socket_error < 0) {
+    $herror = (-($php_socket_error + 100000)) as OS\_Private\HError;
```

Comment:
sneaky change :)

maybe put the 10k/100k into a const somewhere

Reply:
nice catch